polymorphism/README.md

```
@@ -0,0 +1,193 @@
+You can find the definition of `consume` in `src/consume_class_that_adheres_to_concept.ipp` which is then included into `src/consume_class_that_adheres_to_concept.cpp`. In the latter file we also place all explicit instantiations of `consume`:
+
+```c++
+// explicit template instantiation
+template std::string consume<Impl>(Impl&);
+```
+
+As an alternative, the *definition* of `consume` could be in the header `include/polymorphism/consume_class_that_adheres_to_concept.hpp`. Then the explicit template instantiation is no longer required.
+
+## Testing
+
+In this example, I use [µt](https://github.com/boost-ext/ut) as testing framework for no good reason. It looks promising and lightweight and it was written by [Krzysztof Jusiak](https://github.com/krzysztof-jusiak) whom I adore for his outstanding C++ programming skills and his [C++ Tip of The Week](https://github.com/tip-of-the-week/cpp) collection of crazy C++ riddles.   
```

Markdownlint

Expected: 0 or 2; Actual: 3
Trailing spaces

(MD009, no-trailing-spaces)